haystack/generator/transformers.py

```diff
@@ -291,3 +292,181 @@ def predict(self, query: str, documents: List[Document], top_k: Optional[int] =
         result = {"query": query, "answers": answers}
 
         return result
+
+
+class Seq2SeqGenerator(BaseGenerator):
+
+    _model_input_converters: Dict[str, Callable] = dict()
+    """
+        A generic sequence-to-sequence generator based on HuggingFace's transformers.
+
+        Text generation is supported by so called auto-regressive language models like GPT2,
+        XLNet, XLM, Bart, T5 and others. In fact, any HuggingFace language model that extends
+        GenerationMixin can be used by Seq2SeqGenerator.
+
+        Moreover, as language models prepare model input in their specific encoding, each model
+        specified with model_name_or_path parameter in this Seq2SeqGenerator should have an
+        accompanying model input converter that takes care of prefixes, separator tokens etc.
+        By default, we provide model input converters for a few well-known seq2seq language models (e.g. ELI5). 
+        It is the responsibility of Seq2SeqGenerator user to ensure an appropriate model input converter 
+        is either already registered or specified on a per-model basis in the Seq2SeqGenerator constructor.
+
+        For mode details on custom model input converters refer to _BartEli5Converter
+
+
+        See https://huggingface.co/transformers/main_classes/model.html?transformers.generation_utils.GenerationMixin#transformers.generation_utils.GenerationMixin
+        as well as https://huggingface.co/blog/how-to-generate
+
+        For a list of all text-generation models see https://huggingface.co/models?pipeline_tag=text-generation
+
+        **Example**
+
+        ```python
+        |     query = "Why is Dothraki language important?"
+        |
+        |     # Retrieve related documents from retriever
+        |     retrieved_docs = retriever.retrieve(query=query)
+        |
+        |     # Now generate answer from query and retrieved documents
+        |     generator.predict(
+        |        query=query,
+        |        documents=retrieved_docs,
+        |        top_k=1
+        |     )
+        |
+        |     # Answer
+        |
+        |     {'answers': [" The Dothraki language is a constructed fictional language. It's important because George R.R. Martin wrote it."],
+        |      'query': 'Why is Dothraki language important?'}
+        |
+        ```
+    """
+
+    def __init__(
+            self,
+            model_name_or_path: str,
+            input_converter: Optional[Callable] = None,
```

Comment:
@tholor how user will supply these param if creating pipeline via yaml?

Reply:
Yeah, not sure how that works exactly. LMK

Reply:
@oryx1729 You had a similar case recently with the retrievers where we supply a documentstore object. What is the requirement here for saving/loading yamls? Does it have to inherit from `BaseComponent`? 

I think in any case this could only work for now with predefined callables in the haystack namespace (like  `_BartEli5Converter`). We can tackle custom components later, maybe together with custom nodes in https://github.com/deepset-ai/haystack/issues/955